### DISCUSSION TITLE
Midi and Vimix

### DISCUSSION THREAD

**Discussion by MiguelMd13** [brunoherbelin/vimix#116] [Q&A]
Can i use external midi controller to operate the software?

Thanks, its a great software

**Comment by brunoherbelin** [comment]
Hey !

Natively, vimix does not support MIDI.
The OpenSoundControl protocol over network is the prefered way to go (see [documentation](https://github.com/brunoherbelin/vimix/wiki/Open-Sound-Control-API)) 

[Chataigne](https://benjamin.kuperberg.fr/chataigne/en) is a good companion software for vimix: it supports [MIDI and OSC](https://bkuperberg.gitbook.io/chataigne-docs/modules/protocol/midi) and allows creating a router from MIDI to OSC.

There are other MIDI-to-OSC converters that I never tested but might come handy:
https://github.com/llloret/osmid 
https://www.cockos.com/oscii-bot/ 

Alternatively, vimix can also be controlled with the computer keyboard, keypad, joystick or a [tablet app ](https://github.com/brunoherbelin/vimix/wiki/TouchOSC-companion) with the Input Mapping GUI:
![image](https://github.com/brunoherbelin/vimix/assets/60393189/a0d8b2cd-2785-4dfa-ac15-d2848eb60e2e)

**Comment by brunoherbelin** [comment]
Update : Please check out the Chataigne plugin for vimix  #119
